### ISSUE TITLE
aitw数据集在Qwen VL上的评测结果

### ISSUE THREAD

**Issue by jiennyteng** [njucckevin/SeeClick#43]
作者你好，
非常感谢您的工作和数据集，想咨询一下你们在评估QwenVL在sequential Action task的上的表现时是如何构造模型推理时的prompt/instruction呢，感觉直接通过aitw_test.py的代码测试QwenVL，会出现输出格式和action space的定义的格式不同，无法正常评估模型的性能。请问您们在评测时有限制输出格式相关的prompt吗？
下图为本地推理时，qwenVL的输入与输出：
<img width="778" alt="image" src="https://github.com/user-attachments/assets/31849e2e-bd83-402e-8216-81263ade493d">
<img width="778" alt="image" src="https://github.com/user-attachments/assets/15678d7a-b79a-479c-8294-ede728b61912">
<img width="778" alt="image" src="https://github.com/user-attachments/assets/cb243e7f-3ab7-4832-9f84-ac71840385e9">

**Comment by njucckevin** [comment]
您好，感谢对我们工作的关注。
这里可能有些误解。AITW是有训练集的，论文中Qwen-VL和SeeClick的结果，是指将Qwen-VL和SeeClick-pretrain分别在训练集上微调后测试的结果，提供的代码也是给微调后的模型推理准备的。直接zero-shot推理的话Qwen-VL是不行的，更强的模型（Qwen2-VL或者GPT4）大概也需要构建自己的prompt来调试。

**Comment by jiennyteng** [comment]
![image](https://github.com/user-attachments/assets/4b5c7432-3772-43ba-b186-75abea0f7be8)
好的，感谢您的回复，还想再明确一下，您的意思是这个表格中的QwenVL的测试结果，是单独经过AITW数据SFT后的模型推测精度，而SeeClick模型是经过Grounding还有reffering任务增强后再增加AITW数据集进行SFT后的模型测试结果吗？

那么你们这里GPT-4V和ChatGPT-CoT的测试精度又是怎么评估的呢，有相关的文章或者CoT构造的介绍吗？😂

**Comment by njucckevin** [comment]
是的，原论文中SeeClick与Qwen-VL在下游agent任务上性能比较都是在训练-测试setting下进行的，当时的开源VLM也很难zero-shot推理（但是在grounding benchmark ScreenSpot上的性能比较没有训练集相当于zero-shot）。
上方三个闭源模型的结果来自前人的研究中汇报的结果，在5.2.2节的Compared Methods & Evaluation中引用了这三篇工作。
<img width="465" alt="截屏2024-09-13 10 07 37" src="https://github.com/user-attachments/assets/75e4ddbf-fd97-477d-85d1-8e68a91f4654">

**Comment by jiennyteng** [comment]
你好，我按照readme教程中的步䠫，以qwenVL-chat为基础模型，加载预训练的pretrain-model seeclick 后，使用aitw 数据 进行了sft训练。获得了seeclick-aitw的lora模型权重。
<img width="642" alt="image" src="https://github.com/user-attachments/assets/838a6f9d-777d-4352-b41c-755d2d2a82b2">
想咨询一下，我们在运行aitw_test.py的过程中，是否将modelpath设置为seeclick-aitw的lora模型地址，qwenpath设置为qwenvL的模型地址，进行推理呢？
<img width="778" alt="image" src="https://github.com/user-attachments/assets/0c957ae5-3c5d-4fea-9efb-1f47edf7a9cf">
如果是的话，我在运行上述脚本的时候好像出现了一些模型的加载错误。
<img width="1156" alt="image" src="https://github.com/user-attachments/assets/37f0476e-52da-4136-a89c-421c39275353">

**Comment by njucckevin** [comment]
整个流程看起来没什么问题，但我不太清楚您具体的实现过程：
1. finetune和test时用的--qwen_path不太一样？我看两个文件名不太一致，不知道会不会导致问题
2. 这个mismatch的报错似乎是Qwen系列和版本相关的一个问题https://github.com/QwenLM/Qwen/issues/1129，您的Transformer和PEFT版本是什么样的？
不知道是否配置了agent的相应虚拟环境？https://github.com/njucckevin/SeeClick/blob/main/agent_tasks/readme_agent.md#setup-environment

**Comment by jiennyteng** [comment]
好的，我再排查一下相关问题。
看了一下transformer的版本是4.32.0，peft的版本是0.11.1, 不知道是不是这个问题的影响🤔️
还有就是我推理测试还需要配置一个agentstask的独立环境？好像我在采用你们开源的seeclick-aitw模型进行测试时，并没有出现这个报错，不过测试代码上主要是采用AutoModelForCausalLM.from_pretrain进行模型加载的
<img width="756" alt="image" src="https://github.com/user-attachments/assets/bbf60a30-8f71-474f-a302-73945e91a32c">
在进行自己训练的模型推理的时候采用的是AutoPeftModelForCausalLM.from_pretrain进行模型加载的

**Comment by njucckevin** [comment]
AutoPeftModelForCausalLM和AutoModelForCausalLM的选择应该是没问题的。
你这个transformer和peft版本好像确实和那个QwenLM的issue的情况一样。在https://github.com/njucckevin/SeeClick/blob/main/agent_tasks/readme_agent.md 中有配置agentstask独立环境的引导

**Comment by jiennyteng** [comment]
嗯嗯，感谢您的帮助。
应该是transformer和peft版本的问题，建议推理的时候安装transformers=4.36.2 以及peft==0.7.2

**Comment by jiennyteng** [comment]
作者你好，我采用readme中的fintune脚本，进行了qwenVL的sft训练，
<img width="652" alt="image" src="https://github.com/user-attachments/assets/ddb86ab3-3654-4ba6-bbe5-f8ef92da4c28">
但是在模型测试结果中，发现整体指标远低于文章中提到的54%的acc，
目前测试下来大概只有36%的单步执行精度，请问这个是在训练过程中有特别的参数设置吗？

**Comment by njucckevin** [comment]
不太应该？参数我印象里就是用的脚本默认的参数，而且即使有也不应该差这么多。
你能提供更多的信息吗？例如是否冻结了视觉编码器、训练的epoch数、最终在各个分项上的完整结果等。或者你可以先试试用我提供的微调后的[模型](https://huggingface.co/cckevinn/SeeClick-aitw)测试，看看是训练还是测试的问题？

**Comment by jiennyteng** [comment]
嗯嗯，我这边使用微调模型测试结果应该有60%左右的acc，感觉还是训练脚本中的问题，我也重新在看fintune.py中的一些参数设置，目前我复现baseline的过程中，epoch设置为10,learning rate = 3e-5,gradient-accumulation-steps=8,max-length=704,fix_vit=False, 大致也是按这里的默认参数进行训练的。
<img width="512" alt="image" src="https://github.com/user-attachments/assets/19e022e6-fa61-47e5-adfa-81bcd54b223d">

**Comment by njucckevin** [comment]
在[readme_agent.md](https://github.com/njucckevin/SeeClick/blob/main/agent_tasks/readme_agent.md)中, "the fine-tuning scripts are similar to Qwen-VL, except for we use LoRA to fine-tune customized parameters, as in finetune/finetune.py lines 315-327."，这部分启用了自定义的lora参数微调吧？
另外下游任务微调用的基模型是什么？完整训练了10个epoch吗？
